pennylane_orquestra/cli_actions.py

```diff
@@ -0,0 +1,231 @@
+# Copyright 2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module contains utilities and auxiliary functions for using the Orquestra
+Quantum Engine command line interface (CLI).
+"""
+import subprocess
+import time
+import os
+import urllib.request
+import json
+import tarfile
+
+import yaml
+from appdirs import user_data_dir
+
+
+def qe_get(workflow_id, option="workflow"):
+    """Function for getting information via an Orquestra Quantum Engine CLI
+    call.
+
+    This function is mostly used for retrieving workflow related information.
+
+    Args:
+        workflow_id (str): the workflow id for which information will be
+            retrieved
+
+    Kwargs:
+        option (str): The option specified for the ``qe get`` CLI call.
+            Examples include ``workflow`` and ``workflowresult``.
+
+    Returns:
+        str: response message of the CLI call
+    """
+    process = subprocess.Popen(
+        ["qe", "get", option, str(workflow_id)], stdout=subprocess.PIPE, universal_newlines=True
+    )
+    return process.stdout.readlines()
+
+
+def qe_submit(filepath, keep_file=False):
+    """Function for submitting a workflow via a CLI call.
+
+    Handling submission response messages was based on using Orquestra API
+    v1.0.0.
+
+    Args:
+        filepath (str): the filepath for the workflow file
+
+    Keyword Args:
+        keep_file=False (bool): whether or not to keep or delete the workflow
```

Comment:
How come you put in this option? And why is it `False` by default?

I'm guessing it's because these workflow files are just emphemeral, i.e., they are generated by PL-Orquestra plugin, but don't represent a workflow the user would actually want to repeat?

Reply:
The thought here was not to clutter the disk of the user without them knowing that. `qe_submit` is going to be called by the devices in `PennyLane-Orquestra` and a workflow file will be created and submitted under the hood.

> I'm guessing it's because these workflow files are just emphemeral, i.e., they are generated by PL-Orquestra plugin, but don't represent a workflow the user would actually want to repeat?

Exactly. The `keep_file` option will be available to be toggled by passing the same `kwarg` to a device. That way a user can look at the exact workflow file if they wanted to (or re-submit it in the future independent of using PennyLane-Orquestra).